fedimint-client/src/lib.rs

```diff
@@ -1421,6 +1456,75 @@ impl ClientBuilder {
     }
 }
 
+// Sources config from database or from config source specified
+async fn get_config(
+    db: &Database,
+    config_source: Option<ConfigSource>,
+) -> anyhow::Result<ClientConfig> {
+    let mut dbtx = db.begin_transaction().await;
+    let config_res = match dbtx
+        .find_by_prefix(&ClientConfigKeyPrefix)
+        .await
+        .next()
+        .await
+    {
+        Some((_, config)) => Ok(config),
+        None => {
+            let config = match config_source
+                .clone()
+                .ok_or(anyhow!("No config source was provided"))?
+            {
+                ConfigSource::Config(config) => config.clone(),
+                ConfigSource::Invite(invite_code) => {
+                    try_download_config(invite_code.clone(), 100).await?
+                }
+            };
```

Comment:
Deferring fix which because `lightning_reconnect_test` still builds client with config. Will re-enable the restriction with https://github.com/fedimint/fedimint/pull/2855